Basic stream template

tallen116 opened the pull request:
I wanted to add some custom stream template that I have been working on.  I had a need for one and there wasn't one in the main repo.  I tried to follow the standard naming and way the other templates were laid out.

Would love for this to become official!

alessfg commented on README.md:
```diff
@@ -360,10 +360,40 @@ nginx_http_template:
             health_check: max_fails=1 fail_timeout=10s
 
-# Enable creating dynamic templated NGINX stream configuration files.
+# Enable creating dynamic templated NGINX Stream configuration files.
```
I'd tweak this description, your network_stream block (aka reverse proxying block) is compatible with having upstreams, and the stream module does not support web servers. Stream use cases are going to be limited for the most part to reverse proxying / load balancing.

tallen116 replied:
That was a miss on my part.  Does this sound ok?

# Comment out upstreams depending on whether you wish to create a reverse proxy
# or load balancer configuration file.

alessfg replied:
That could work, although I would delete that whole sentence since in the end even if you use upstreams you are load balancing by acting as a reverse proxy. The only difference between using upstreams or not is whether you want to load balance or not.

alessfg commented on defaults/main.yml:
```diff
@@ -194,5 +194,36 @@ nginx_http_template:
             health_check: max_fails=1 fail_timeout=10s
 
 # Enable creating dynamic templated NGINX stream configuration files.
+# Defaults will not produce a valid configuration. Instead they are meant to showcase
```
See previous comment re the stream block description

alessfg commented on defaults/main.yml:
```diff
@@ -194,5 +194,36 @@ nginx_http_template:
+          proxy_pass: backend
+          proxy_timeout: 3s
+          proxy_connect_timeout: 1s
+          proxy_protocol: "off"
```
```suggestion
          proxy_protocol: false
```

tallen116 replied:
I will have to update the logic in the jinja2 template to accept true and false.  I will work on that.

alessfg replied:
Thanks!

tallen116 commented:
What is your goal with the Travis test files?  Are you testing for playbook and/or ansible run time errors?

tallen116 commented:
Ok totally agree.  I just wanted to make sure we were on the same page on the testing :wink:

tallen116 commented:
I added the test playbook and added to the .travis.yml file.  I also added an option for udp streams as another variable.  Let me know how the testing goes or anything you want changed.

alessfg commented:
I made some final changes to bring stream templating more in line with http templating. Everything else LGTM!
